generator/feature_helpers.hpp

```diff
@@ -0,0 +1,103 @@
+#pragma once
+
+#include "geometry/distance.hpp"
+#include "geometry/point2d.hpp"
+#include "geometry/simplification.hpp"
+
+#include "indexer/coding_params.hpp"
+#include "indexer/scales.hpp"
+
+#include "base/assert.hpp"
+#include "base/math.hpp"
+
+#include <cmath>
+#include <cstdint>
+#include <limits>
+#include <vector>
+
+class FeatureBuilder1;
+
+namespace feature
+{
+class CalculateMidPoints
+{
+public:
+  using CellAndOffset = std::pair<uint64_t, uint64_t>;
+
+  CalculateMidPoints() = default;
+
+  void operator()(FeatureBuilder1 const & ft, uint64_t pos);
+  bool operator()(m2::PointD const & p);
+
+  m2::PointD GetCenter() const;
+  std::vector<CellAndOffset> const & GetVector() const { return m_vec; }
+
+  void Sort();
+
+private:
+  m2::PointD m_midLoc;
+  m2::PointD m_midAll;
+  size_t m_locCount = 0;
+  size_t m_allCount = 0;
+  uint32_t m_coordBits = serial::CodingParams().GetCoordBits();
+  std::vector<CellAndOffset> m_vec;
+};
+
+template <class Point>
+inline bool ArePointsEqual(Point const & p1, Point const & p2)
+{
+  return p1 == p2;
+}
+
+template <>
+inline bool ArePointsEqual<m2::PointD>(m2::PointD const & p1, m2::PointD const & p2)
+{
+  return AlmostEqualULPs(p1, p2);
+}
+
+class BoundsDistance : public m2::DistanceToLineSquare<m2::PointD>
+{
+public:
+  BoundsDistance(m2::RectD const & rect) : m_rect(rect), m_eps(5.0E-7)
```

Comment:
explicit?

Reply:
Opt. Why not moving initialization of m_eps to declaration?

Reply:
done